server/tests/src/main/scala/sttp/tapir/server/tests/ServerRejectTests.scala

```diff
@@ -40,7 +41,10 @@ class ServerRejectTests[F[_], ROUTE](
     testServer(endpoint.get.in("path"), "should return 404 for an unsupported method, when a single endpoint is interpreted")((_: Unit) =>
       pureResult(().asRight[Unit])
     ) { (backend, baseUri) =>
-      basicRequest.post(uri"$baseUri/path").send(backend).map(_.code shouldBe StatusCode.NotFound)
+      val statusCode =
+        if (useMethodNotAllowedForUnsupportedMethod) StatusCode.MethodNotAllowed
```

Comment:
I removed HTTP methods specified in an `Endpoint` from `Route` of Armeria. 
All HTTP methods are allowed at Armeria level so that `RejectInterceptor` customizes the response of invalid requests.